src/classes/AwsProvider.ts

```diff
@@ -84,6 +85,35 @@ export class AwsProvider {
         this.serverless.service.setFunctionNames(this.serverless.processedInput.options);
     }
 
+    /**
+     * @internal
+     */
+    setVpcConfig(securityGroup: AwsCfInstruction, subnets: AwsCfInstruction[]): void {
+        if (this.getVpcConfig() !== null) {
+            throw new ServerlessError(
+                "Can't register more than one VPC.\n" +
+                    'Either you have several "vpc" constructs \n' +
+                    'or you already defined "provider.vpc" in serverless.yml',
+                "LIFT_ONLY_ONE_VPC"
+            );
+        }
+
+        this.serverless.service.provider.vpc = {
+            securityGroupIds: [securityGroup], // TODO : merge with existing groups ?
```

Comment:
I think the TODO can be removed now? (if VPC settings are defined in serverless.yml then there will be an error)

Reply:
That's more of an idea for later. 

Security groups are not related to a specific subnet/vpc so one may want to provide a custom security group that is shared accross projets/environments and we might want to merge our newly created one with those provided by the user.

Reply:
👍  ok makes sense